xymol/xymol.py

```diff
@@ -133,3 +134,35 @@ def plot_similarity_map(
             self.mol, weights, **kwargs
         )
         img.savefig(file_name, bbox_inches='tight', dpi=600)
+
+    def create_map(
+        self, featurizer, model, filename="similarity_map.png"
+        ) -> list:
+        """Create similarity map
+
+        Args:
+            featurizer (object): must have a featurize() function.
+            model (object): must have a predict() function.
+
+        Returns:
+            list: a list of weights
+        """
+        if not( hasattr(featurizer, "featurize") and callable(getattr(featurizer, "featurize")) ):
+            raise ValueError("Featurizer must have featurize() function.")
+
+        if not( hasattr(model, "predict") and callable(getattr(model, "predict")) ):
+            raise ValueError("Model must have predict() function.")
+
+
+        parent_feat = np.array(featurizer.featurize(self.smiles))
+        parent_prediction = model.predict(parent_feat)[0]
+
+        drop_atoms = np.array(self.drop_each_atom())
+        drop_atoms_feat = np.array(featurizer.featurize(drop_atoms))
+
+        predictions = model.predict(drop_atoms_feat)
+
+        weights = [pred - parent_prediction for pred in predictions]
+        self.plot_similarity_map(weights, filename)
+
+        return weights
```

Comment:
we probably do not need to return weights. This API is simply used for plotting.

Reply:
OK, should I also remove these lines in the test code?
```   
    # check that all values in weights equal correct test difference (drop_atom pred minus parent_pred)
    assert weights[0] == TEST_WEIGHT
    assert weights.count(weights[0]) == len(weights)
```

Reply:
yes. Just try to check that there is a new png file.